libs/pika/async_mpi/tests/unit/algorithm_transform_mpi.cpp

```diff
@@ -234,19 +229,33 @@ int pika_main()
         }
     }
 
-    test_adl_isolation(mpi::transform_mpi(my_namespace::my_sender{}, [](MPI_Request) {}));
+    test_adl_isolation(mpi::transform_mpi(my_namespace::my_sender{}, [](MPI_Request*) {}));
```

Comment:
I think I missed this change in another PR: does this need to be passed by pointer? I'd prefer if we did it by reference.

Reply:
actuaally, dispatch passes a request to trigger, which returns nothing, so this is wrong, but when I remove the args from the lambda, compilation fails. not quite sure what the problem is.

Reply:
Well, you definitely  need _an_ argument to the lambda. The question is what type of request it should be. The lambdas to other `transform_mpi` calls also take the request by pointer, so it seems to be consistent. I'm just wondering why it needs to be a pointer (rather than reference, or even by value)?

Reply:
After discussion this is indeed the correct fix. The callables take a pointer to a request to mirror what MPI functions do. This test was not using the correct argument type, and it's possible that we were previously getting some implicit conversion or similar that hid this mistake.

Thanks @biddisco for fixing this!